### ISSUE TITLE
Redis : laravel connection refused [tcp://127.0.0.1:6379]

### ISSUE THREAD

**Issue by aeq-dev** [dunglas/frankenphp#448]
Hello,
 I'm using redis for cache, I'm getting this error : 
`laravel connection refused [tcp://127.0.0.1:6379]`

**Comment by dunglas** [comment]
Indeed, this looks more like a configuration problem on your Redis server. Could you provide more details please?

**Comment by aeq-dev** [comment]
> Indeed, this looks more like a configuration problem on your Redis server. Could you provide more details please?

Here's it : 
```
REDIS_HOST=localhost // Even if I set redis
REDIS_PASSWORD=null
REDIS_PORT=6379

```

**Comment by mxmp210** [comment]
If you're running it inside container, it cannot reach main host by default as they are on different network subnets. "localhost" refers to frankenphp container itself, you'll either need to open network or bring in service to same network. More context would be appreciated. It looks like config issue rather than runtime error.

**Comment by dellow** [comment]
I'm getting the same thing, even for an external Redis cache server. Seems something is going on here.

Frankenphp used via Octane installed via Laravel Forge. If I switch to use Swoole instead works straight away. Switch back to Frankenphp and it doesn't.

**Comment by mxmp210** [comment]
> I'm getting the same thing, even for an external Redis cache server. Seems something is going on here.
> 
> Frankenphp used via Octane installed via Laravel Forge. If I switch to use Swoole instead works straight away. Switch back to Frankenphp and it doesn't.

AFAIK forge allows you to have cache server type, in that case the instance is separate from your main app, in those cases you need to have them on same vpcs / subnets and let server connect, it would provision server on same network by default but depending on your deployment and network config, server might not be reachable.

Not sure how forge handles frankenphp process i.e. as standalone process in machine eith custom script or as docker process - as it's more of typical fast cgi web server config wrapper and automates most of the infrastructure config in an opinionated way with ssh interaction.

Sail on other hand uses docker as backend and this explains why you're not able to connect redis as it might be on other server and docker being in bridge mode can only see servers in it's own subnets. 

Check IP ranges, make sure they share same parent subnets or your network is able to find server ( via routing table mapped - out of the scope for managed services like sail or forge ).

**Comment by flexchar** [comment]
@aeq-dev can you check if you an access redis as such using `nc -vz localhost 6379` from exactly the environment where your Laravel is exepcted to run?

Then check if you see any unusual hosts mapped in `/etc/hosts` so `cat /etc/hosts` and share with us.

This sounds a lot like a not related issue to me but Docker can be confusing at times.

**Comment by SuheylZ** [comment]
I'm facing exactly the same issue. i run frankenphp outside as a binary, it runs simple scripts fine but as i try to correct with redis or kafka, connection is refused.

**Comment by henderkes** [comment]
Is the redis extension installed and your server reachable? I can confirm that it's working for me, so this isn't a FrankenPHP issue.

**Comment by SuheylZ** [comment]
i do not understand this "working on my machine". this is the same reason we have containerized apps. if there are so many peeps reporting the issues then there must be the issue. Franken php cannot contact to redis with authentication enabled. even though Simone php is able to connect, dockerized version of script also works but fraken php, the connection is simple refused.

**Comment by henderkes** [comment]
The frankenphp binary does not contain support for ssl/sasl yet, static-php-cli only added it recently: https://github.com/crazywhalecc/static-php-cli/pull/955/files . It will automatically be added in the next release. If it's not working in Docker, there is nothing we can do, because it does not relate to frankenphp, but the redis extension itself. We rely on the official Docker images.

**Comment by SuheylZ** [comment]
sorry if i wasnt clear enough

redis (no auth)
- php cli ok
- franken php ok
- docker (php cli) ok

redis ( auth)
- php cli ok
- franken php ERROR
- docker (php cli) ok

**Comment by henderkes** [comment]
The static binary will work with auth with the next release (presumably v1.10.0).
